File: src/GradeMe3/Projects/Project.cs
using System.Collections.Generic;
using GradeMe3.Courses;
using GradeMe3.Evaluations;
using GradeMe3.Students;
using GradeMe3.Teams;
using Newtonsoft.Json;

namespace GradeMe3.Projects
{
    public class Project
    {
        public string Id { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public string EmailTemplate { get; set; }
        public string Status { get; set; }
        public string CourseId { get; set; }
        [JsonIgnore]
        public Course Course { get; set; }
        [JsonIgnore]
        public List<Evaluation> Evaluations { get; set; }
        [JsonIgnore]
        public List<Team> Teams { get; set; }
        [JsonIgnore]
        public List<StudentInProject> StudentsInProject { get; set; }
    }
}

File: src/server/entities/Project.ts
import {Column, Entity, OneToMany, PrimaryColumn} from 'typeorm';
import {Evaluation} from './Evaluation';
import {Course} from './Course';

@Entity()
export class Project {
    @PrimaryColumn()
    public id: string;
    @OneToMany(type => Course, course => course.projects)
    public course: Course;
    @Column()
    public name: string;
    @Column()
    public description: number;
    @Column()
    public status: string;
    @Column()
    public emailTemplate: string;
    @OneToMany(type => Evaluation, evaluation => evaluation.project)
    public evaluations: Evaluation[];
}